tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyBulkClosePosts.java

```diff
@@ -0,0 +1,124 @@
+package org.unicode.cldr.web;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+
+import org.json.JSONException;
+import org.unicode.cldr.util.XMLSource;
+import org.unicode.cldr.web.SurveyAjax.JSONWriter;
+
+public class SurveyBulkClosePosts {
+
+    private SurveyMain sm;
+
+    private boolean execute;
+
+    /**
+     * Construct a SurveyBulkClosePosts
+     *
+     * @param sm the SurveyMain
+     * @param execute if true, actually close the posts; else report count and provide button
```

Comment:
this is so close to being a REST API… in my mind I can almost see the @-annotations on each function!

Reply:
There really should be a progress bar, since the db queries could take a minute or more, for thousands of posts. Let's decide on a uniform way to implement progress bars for all potentially long-running server tasks. Cf. https://unicode-org.atlassian.net/browse/CLDR-14251